riptide-stream/src/main/java/org/zalando/riptide/stream/StreamSpliterator.java

```diff
@@ -47,6 +47,7 @@ public boolean tryAdvance(Consumer<? super T> action) {
         try {
             JsonToken token = parser.nextToken();
             if (token == null) {
+                parser.close();
```

Comment:
Ok, so a Stream itself is Closeable, which means there has to be a way to hook into it. I read a little bit about it and it looks like terminal operations (forEach, collect, ...) actually do **not** close the stream. So in general clients should always close them.